src/toil/jobStores/aws/jobStore.py

```diff
@@ -1153,39 +1176,114 @@ def __getBucketVersioning(self, bucket):
         for attempt in retry_s3():
             with attempt:
                 status = bucket.get_versioning_status()
-        return bool(status) and self.versionings[status['Versioning']]
+        return bool(status) and cls.versionings[status['Versioning']]
 
-    def __getBucketRegion(self, bucket):
+    @staticmethod
+    def __getBucketRegion(bucket):
         for attempt in retry_s3():
             with attempt:
                 return bucket_location_to_region(bucket.get_location())
 
     def deleteJobStore(self):
-        self.registry_domain.put_attributes(self.namePrefix, dict(exists=str(False)))
-        if self.filesBucket is not None:
-            for attempt in retry_s3():
-                with attempt:
-                    for upload in self.filesBucket.list_multipart_uploads():
-                        upload.cancel_upload()
-            if self.__getBucketVersioning(self.filesBucket) in (True, None):
-                for attempt in retry_s3():
-                    with attempt:
-                        for key in list(self.filesBucket.list_versions()):
-                            self.filesBucket.delete_key(key.name, version_id=key.version_id)
-            else:
+        self.cleanJobStore(self.locator)
+
+    @classmethod
+    def cleanJobStore(cls, locator):
+        existed = False
+        failure = False
+        with closing(cls._connectS3(locator.region)) as s3:
+            try:
                 for attempt in retry_s3():
                     with attempt:
-                        for key in list(self.filesBucket.list()):
-                            key.delete()
-            for attempt in retry_s3():
-                with attempt:
-                    self.filesBucket.delete()
-        for domain in (self.filesDomain, self.jobsDomain):
-            if domain is not None:
+                        filesBucket = s3.get_bucket(locator.namePrefix +
+                                                    cls.nameSeparator +
+                                                    'files')
+                        existed = True
+            except S3ResponseError as e:
+                if e.error_code == 'NoSuchBucket':
+                    filesBucket = None
+                else:
+                    raise
+            i, maxAttempts = 0, 5
```

Comment:
We don't need another retry logic. You have to get by without log_component_failure() and without this loop I'm afraid.